test/src/test/java/mil/nga/giat/geowave/test/basic/AbstractGeoWaveBasicVectorIT.java

```diff
@@ -218,6 +220,129 @@ protected void testDelete(
 				success);
 	}
 
+	protected void testBulkDeleteSpatial(
+			final URL savedFilterResource,
+			final PrimaryIndex index )
+			throws Exception {
+		LOGGER.warn("bulk deleting from " + index.getId() + " index");
+
+		final mil.nga.giat.geowave.core.store.DataStore geowaveStore = getDataStorePluginOptions().createDataStore();
+		final DistributableQuery query = TestUtils.resourceToQuery(savedFilterResource);
+		CloseableIterator<?> actualResults;
+
+		actualResults = geowaveStore.query(
```

Comment:
to test dupe bookkeeping, use a null query to count all the results as well as this to count how many should be deleted...after delete query all the results again and make sure it equals `initialAllResults - expectedFeaturesDeleted`